core/tests/performance_framework/selenium_driver.py

```diff
@@ -0,0 +1,231 @@
+    contains an entry for every resource required to load and display a
+    web-page. Each entry contains the url requested, request and response
+    headers.
+    Refer: https://dvcs.w3.org/hg/webperf/raw-file/tip/specs/HAR/Overview.html
+    """
+
+    def __init__(self, browser='chrome'):
+        if browser in ['chrome', 'firefox']:
+            self.browser = browser
+        else:
+            raise ValueError
+
+        if self.browser == 'chrome':
+            os.environ["webdriver.chrome.driver"] = CHROME_DRIVER_PATH
+
+        self.chrome_driver_path = CHROME_DRIVER_PATH
+        self.browser_mob_path = BROWSERMOB_PROXY_PATH
+
+        self.server = self.driver = self.proxy = None
+
+    def get_har_dict(self, page_url):
+        """Retrieve HTTP Archive (HAR) or page session stats for a page URL.
+
+        To get it we require the use of a proxy server as we need to record the
+        communication between server and client.
+        """
+        # pylint: disable=unused-variable
+        with Xvfb() as xvfb:
+        # pylint: enable=unused-variable
+            self._setup_proxy()
+            self._setup_driver(use_proxy=True)
+
+            self.proxy.new_har(page_url, options={'captureHeaders': True})
+
+            self.driver.get(page_url)
+
+        har_dict = self.proxy.har
+
+        self._clear_proxy()
+        self._clear_driver()
+
+        return har_dict
+
+    def get_har_dict_cached_state(self, page_url):
+        """Retrieve HTTP Archive (HAR) or page session stats for a page URL
+        while simulating a cached state i.e, a return user.
+
+        To get it we require the use of a proxy server as we need to record the
+        communication between server and client.
+        """
+        # pylint: disable=unused-variable
+        with Xvfb() as xvfb:
+        # pylint: enable=unused-variable
+            self._setup_proxy()
+            self._setup_driver(use_proxy=True)
+
+            # Fetch the page once. This leads to caching of various resources.
+            self.driver.get(page_url)
+
+            # Start recording har data for the next page fetch.
+            self.proxy.new_har(page_url, options={'captureHeaders': True})
+            self.driver.get(page_url)
+
+        har_dict = self.proxy.har
+
+        self._clear_proxy()
+        self._clear_driver()
+
+        return har_dict
+
+    def get_page_session_timings(self, page_url):
+        """Retrieve page load timings for a page URL."""
+        # pylint: disable=unused-variable
+        with Xvfb() as xvfb:
+        # pylint: enable=unused-variable
+            self._setup_driver(use_proxy=False)
+
+            self.driver.get(page_url)
+
+            # To get entries for all the resources we can use:
+            #   driver.execute_script("return window.performance.getEntries();")
+            # To get only the timing we can use:
+            #   driver.execute_script("return window.performance.timing;")
```

Comment:
Are these comments useful? They aren't providing additional context. Prefer removing them (here and elsewhere) if they aren't going to eventually become code. Otherwise, please turn them into code and remove the comments.